test/unit/strings_spec.lua

```diff
@@ -98,3 +98,23 @@ describe('vim_strnsave_unquoted()', function()
     eq('/Program\\nFiles/sh', vim_strnsave_unquoted('/Program"\\n"Files/sh'))
   end)
 end)
+
+describe("reverse_text", function()
+  local reverse_text = function(str)
+    return helpers.internalize(strings.reverse_text(to_cstr(str)))
+  end
+
+  it("handles empty string", function()
+    eq("", reverse_text(""))
+  end)
+
+  it("handles simple cases", function()
+    eq("a", reverse_text("a"))
+    eq("ba", reverse_text("ab"))
+  end)
+
+  it("handles multibyte characters", function()
+    eq("bα", reverse_text("αb"))
+    eq("Yötön yö", reverse_text("öy nötöY"))
+  end)
```

Comment:
I would also try with combining chars (that don't exists precomposed), like p͓̚ , maybe written as "aap\xcc\x9a\xcd\x93bb" as some terminals won't render it correctly.

Also some string with invalid/incomplete utf like "ab\xe0", it might be "garbage in/garbage out" (or not even reversible, no pun intended), but it shouldn't be "garbage in/nvim crashes" (AFAIK `mb_ptr2len` already handles this as sanely as possible, but more tests for it wouldn't hurt)